controllers/clustertopology_controller_compute_test.go

```diff
@@ -413,11 +415,11 @@ func TestObjToRef(t *testing.T) {
 }
 
 var (
-	fakeInfrastructureProviderGroupVersion = schema.GroupVersion{Group: "infrastructure.cluster.x-k8s.io", Version: "v1alpha4"}
-	fakeControlPlaneProviderGroupVersion   = schema.GroupVersion{Group: "controlplane.cluster.x-k8s.io", Version: "v1alpha4"}
-	fakeBootstrapProviderGroupVersion      = schema.GroupVersion{Group: "bootstrap.cluster.x-k8s.io", Version: "v1alpha4"}
+	FakeInfrastructureProviderGroupVersion = schema.GroupVersion{Group: "infrastructure.cluster.x-k8s.io", Version: "v1alpha4"}
```

Comment:
My assumption is that we should drop those group versions and related CRDs in favour of the ones defined in internal/testtypes.
@sbueringer to confirm here

Reply:
This PR in its current state really only shows the file structure and basic idea of what's being proposed in  #5055 This first commit is pretty much just a copy and paste, so the types here aren't fixed, and they're not even really suggestions. Part of the overall effort will be to rationalise the CRDs used for testing as in  #5040. I'll add commits to this PR in the next few days that show more of the design intention.